GVFS/GVFS.UnitTests/Common/EpochConverterTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿using GVFS.Common;
+using GVFS.Tests.Should;
+using NUnit.Framework;
+using System;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class EpochConverterTests
+    {
+        [TestCase]
+        public void DateToEpochToDate()
+        {
+            DateTime time = new DateTime(2018, 12, 18, 8, 12, 13, DateTimeKind.Utc);
+            DateTime converted = EpochConverter.FromUnixEpochSeconds(EpochConverter.ToUnixEpochSeconds(time));
+
+            time.ShouldEqual(converted);
+        }
+
+        [TestCase]
+        public void EpochToDateToEpoch()
+        {
+            long time = 15237623489;
+            long converted = EpochConverter.ToUnixEpochSeconds(EpochConverter.FromUnixEpochSeconds(time));
+
+            time.ShouldEqual(converted);
+        }
+
+        [TestCase]
+        public void FixedDates()
+        {
+            DateTime[] times = new DateTime[]
+            {
+                new DateTime(2018, 12, 13, 20, 53, 30, DateTimeKind.Utc),
+                new DateTime(2035, 1, 3, 5, 0, 59, DateTimeKind.Utc),
+                new DateTime(1989, 12, 31, 23, 59, 59, DateTimeKind.Utc)

```

Comment:
Should we also test that the subsequent second results in the same epoch?

Reply:
Mostly I was just making up an old date. Do you think the year boundary matters?

Reply:
You managed to make up a date within one second of a leap second, so i figured that was intentiona. I think we should have a unit test that we handle leap seconds correctly.

Reply:
Also, I think my previous comment was incorrect, the ambiguous datetime I think would be 1989/12/31 23:59:60 with 1990/01/01 00:00:00

Reply:
It's actually the 1990/1991 boundary, but `DateTime` doesn't register these as two seconds apart.

Reply:
https://en.wikipedia.org/wiki/Leap_second